pkg/controllers/cloudprovideraccount/controller.go

```diff
@@ -175,70 +177,65 @@ func (r *CloudProviderAccountReconciler) setSyncStatusAndSecretWatcher() {
 
 // getCpaBySecret returns nil only when the Secret is not used by any CloudProvideAccount CR,
 // otherwise the dependent CloudProvideAccount CRs will be returned.
-func (r *CloudProviderAccountReconciler) getCpaBySecret(s types.NamespacedName) ([]crdv1alpha1.CloudProviderAccount, error) {
+func (r *CloudProviderAccountReconciler) getCpaBySecret(s *types.NamespacedName) ([]*crdv1alpha1.CloudProviderAccount, error) {
 	cpaList := &crdv1alpha1.CloudProviderAccountList{}
 	if err := r.Client.List(context.TODO(), cpaList, &client.ListOptions{}); err != nil {
 		return nil, fmt.Errorf("failed to get CloudProviderAccount list, err %v", err)
 	}
-	var cpaItems []crdv1alpha1.CloudProviderAccount
+	var cpaItems []*crdv1alpha1.CloudProviderAccount
 	for _, cpa := range cpaList.Items {
 		if cpa.Spec.AWSConfig != nil {
 			if cpa.Spec.AWSConfig.SecretRef.Name == s.Name &&
 				cpa.Spec.AWSConfig.SecretRef.Namespace == s.Namespace {
-				cpaItems = append(cpaItems, cpa)
+				cpaItems = append(cpaItems, &cpa)
 			}
 		}
-
 		if cpa.Spec.AzureConfig != nil {
 			if cpa.Spec.AzureConfig.SecretRef.Name == s.Name &&
 				cpa.Spec.AzureConfig.SecretRef.Namespace == s.Namespace {
-				cpaItems = append(cpaItems, cpa)
+				cpaItems = append(cpaItems, &cpa)
 			}
 		}
 	}
 	return cpaItems, nil
 }
 
-// watchSecret watch the Secret objects.
+// updateDependentCpaBySecretNamespace updates all dependent accounts for a given Secret Namespace.
+func (r *CloudProviderAccountReconciler) updateDependentCpaBySecretNamespace(secretNamespacedName *types.NamespacedName,
+	eventType watch.EventType) {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+
+	cpaItems, err := r.getCpaBySecret(secretNamespacedName)
+	if err != nil {
+		r.Log.WithName("Secret").Error(err, "error getting CPA by Secret", "Secret", secretNamespacedName)
+	}
+
+	for _, cpa := range cpaItems {
+		accountNamespacedName := &types.NamespacedName{Namespace: cpa.Namespace, Name: cpa.Name}
+		// For add operation, update CPA only when status contains an error ErrorMsgSecretDoesNotExist.
```

Comment:
there is no handling for delete, how are you setting ErrorMsgSecretDoesNotExist in CPA?